src/components/TagManagementModal/tag-management-service.ts

```diff
@@ -0,0 +1,91 @@
+import {
+	ResourceTagBase,
+	ResourceTagInfo,
+	ResourceTagSubmitInfo,
+	SubmitInfo,
+	TaggedResource,
+} from './models';
+import keys from 'lodash/keys';
+import flatMap from 'lodash/flatMap';
+import map from 'lodash/map';
+import groupBy from 'lodash/groupBy';
+
+export const ALL_TAGS_COLUMN_KEY = 'All Tags';
+
+export const getTagKeyValueComposite = (tagKey: string, value: string) =>
+	`${tagKey}: ${value}`;
+
+export const getResourceTags = <T extends {}, P extends keyof T>(
+	item: T,
+	tagField: P,
+) => ((tagField in item ? item[tagField] : null) ?? []) as ResourceTagBase[];
+
+export const groupResourcesByTags = <
+	T extends TaggedResource,
+	P extends keyof T,
+>(
+	items: T[],
+	tagField: P,
+) => {
+	const resourceTagInfos = flatMap(items, (item) => {
+		const tags = getResourceTags(item, tagField);
+		return tags.map((tag: any) => ({
```

Comment:
```suggestion
		return tags.map((tag) => ({
```
`groupResourcesByTags` already returns properly typed results.